Update docker-compose.yml

beankylla opened the pull request:
Was getting following error:
Failed to write into /var/log/airdcpp/2022-02-system.log: No such file or directory.
Solved with this adjustment

beankylla committed:
Update docker-compose.yml

gangefors commented:
Thanks for the PR, but this is a change that is not valid for the default configuration of the program.

The [default log directory](https://github.com/airdcpp-web/airdcpp-webclient/blob/b064095c5480af2dec2e480e911a20849421258b/airdcpp-core/airdcpp/SettingsManager.cpp#L430) is ` /airdcpp-webclient/config/Logs/` and it is created on launch for a fresh install.
```
$ docker run --name adccontainer --rm -d -e PUID=1000 -e PGID=1000 -p 5600:5600 gangefors/airdcpp-webclient:latest
da4f2a49fa583145947c412b2a7d2ee6cf29acb2baecbdbf4c0bd9d13cd1da9a

$ docker exec adccontainer stat /airdcpp-webclient/config/Logs/
  File: /airdcpp-webclient/config/Logs/
  Size: 4096            Blocks: 8          IO Block: 4096   directory
Device: 46h/70d Inode: 197918      Links: 2
Access: (0755/drwxr-xr-x)  Uid: ( 1000/ airdcpp)   Gid: ( 1000/ airdcpp)
Access: 2022-02-26 15:00:14.561822091 +0000
Modify: 2022-02-26 15:00:14.561822091 +0000
Change: 2022-02-26 15:00:14.561822091 +0000
 Birth: 2022-02-26 15:00:14.561822091 +0000
```

If you need a different configuration then you should modify your local docker-compose.yml file. The one provided in the git should be valid for an unconfigured fresh install, which it is.

beankylla commented:
hi thanks for answering  so completely and so quickly.
I'm just wondering one thing. I was using those default settings on docker but getting the error. Was I doing anything wrong?

gangefors commented:
If i start the application using `docker-compose up` after cloning the git I get the following output.

```bash
$ docker-compose up
Starting docker-airdcpp-webclient_airdcpp_1 ... done
Attaching to docker-airdcpp-webclient_airdcpp_1
airdcpp_1  | + is_podman
airdcpp_1  | + [[ '' == \p\o\d\m\a\n ]]
airdcpp_1  | + [[ -f /run/.containerenv ]]
airdcpp_1  | ++ id -u
airdcpp_1  | + [[ 0 -ne 0 ]]
airdcpp_1  | + validate_id
airdcpp_1  | + [[ -z 1000 ]]
airdcpp_1  | + [[ -z 1000 ]]
airdcpp_1  | + [[ 1000 -lt 101 ]]
airdcpp_1  | + [[ 1000 -lt 101 ]]
airdcpp_1  | + create_user_and_group
airdcpp_1  | ++ id -u airdcpp
airdcpp_1  | + [[ '' != \1\0\0\0 ]]
airdcpp_1  | + groupdel airdcpp
airdcpp_1  | + userdel airdcpp
airdcpp_1  | + groupadd -f -g 1000 airdcpp
airdcpp_1  | + useradd -u 1000 -g 1000 --no-create-home -s /usr/sbin/nologin airdcpp
airdcpp_1  | + take_ownership
airdcpp_1  | + chown -R 1000:1000 /.airdcpp
airdcpp_1  | + chmod -R u+rw /.airdcpp
airdcpp_1  | + init_config
airdcpp_1  | + [[ ! -r /.airdcpp/DCPlusPlus.xml ]]
airdcpp_1  | + rm -f /.airdcpp/WebServer.xml.bak
airdcpp_1  | + exec runuser -u airdcpp -g airdcpp -- /airdcpp-webclient/airdcppd
airdcpp_1  | Starting.
airdcpp_1  | .Loading Hash database
airdcpp_1  | Loading Download queue
airdcpp_1  | Loading Shared files
airdcpp_1  | Loading Country information
airdcpp_1  | Loading Connectivity
airdcpp_1  | Loading Web server
airdcpp_1  | .
airdcpp_1  |
airdcpp_1  | AirDC++w 2.11.2 running, press ctrl-c to exit...
airdcpp_1  |
airdcpp_1  | HTTP port: 5600, HTTPS port: 5601
airdcpp_1  | Config path: /airdcpp-webclient/config/
airdcpp_1  | Web resources path: /airdcpp-webclient/web-resources/
airdcpp_1  | [2022-02-27 08:32:37] [http] localhost:5600 [::1]:33620 "GET /index.html HTTP/1.1" 200 2841 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
airdcpp_1  | [2022-02-27 08:32:37] [http] localhost:5600 [::1]:33622 "GET /favicon.ico HTTP/1.1" 404 40 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
airdcpp_1  | [2022-02-27 08:32:37] [error] handle_read_handshake error: websocketpp.transport:7 (End of File)
airdcpp_1  | [2022-02-27 08:32:37] [error] handle_read_handshake error: websocketpp.transport:7 (End of File)
airdcpp_1  | [2022-02-27 08:32:37] [fail] WebSocket Connection [::1]:33628 - "" - 0 websocketpp.transport:7 End of File
airdcpp_1  | [2022-02-27 08:32:37] [fail] WebSocket Connection [::1]:33626 - "" - 0 websocketpp.transport:7 End of File
airdcpp_1  | [2022-02-27 08:32:37] [http] localhost:5600 [::1]:33624 "GET /images/background_winter_3840px.0e07df195a857a526476.jpg HTTP/1.1" 200 633714 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
airdcpp_1  | [2022-02-27 08:32:38] [http] localhost:5600 [::1]:33630 "GET /sw.js HTTP/1.1" 200 1510 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
airdcpp_1  | [2022-02-27 08:32:43] [connect] WebSocket Connection [::1]:33632 v13 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0" /api/v1/ 101
airdcpp_1  | [2022-02-27 08:32:43] [http] localhost:5600 [::1]:33634 "GET /images/AirDCPlusPlus.1eb3f49b41aea45df8aa.png HTTP/1.1" 200 19944 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
airdcpp_1  | [2022-02-27 08:32:43] [http] localhost:5600 [::1]:33638 "GET /proxy?url=https%3A%2F%2Fregistry.npmjs.org%2F-%2Fv1%2Fsearch%3Ftext%3Dkeywords%3Aairdcpp-extensions-public%26size%3D100 HTTP/1.1" 200 8344 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
airdcpp_1  | [2022-02-27 08:32:43] [http] localhost:5600 [::1]:33636 "GET /proxy?url=https%3A%2F%2Fairdcpp-web.github.io%2Ffeed.xml HTTP/1.1" 200 37967 "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"
```
And this is what shows on the Settings>System>Logging page after logging in.
![image](https://user-images.githubusercontent.com/289708/155875001-707ccd6a-ec2a-4429-b279-eab4187f09cc.png)

I can't see any indication that `/var/log/airdcpp` is used anywhere.

Can you please submit your output and go into details where you encounter the error?

gangefors commented:
One thing I came to think of is that the volume that docker-compose creates isn't removed just because you recreate the container. Remove all resources by running `docker-compose down --volumes` before launching again with `docker-compose up`.

The volume contains the application configuration and in your case it might still point the log directory to `/var/log/airdcpp`.

Or you could just update the log path setting to `/airdcpp-webclient/config/Logs/` and go about your day.

beankylla commented:
Thanks a lot for the support! Now i think i know where this comes from, sorry for the trouble!
I initially had airdcpp natively installed, and then migrated to Docker. At that moment i moved the configuration which probably included the log directory.
Just modified that to point to somewhere in the config directory so all good.
Sorry for needing help to figure this out and thanks a LOT for the packaging really appreciating it ! :)

gangefors commented:
Happy to help.
And yes, migrating a configuration would be able to cause of the error you encountered.

For the future, when reporting issues to projects, be sure to replicate the issue in a clean environment and always attach the complete list of commands used and logs/scheenshots that shows the error. This makes it easier for project owners to figure out the problem.
